api/src/main/java/com/velocitypowered/api/proxy/ProxyServer.java

```diff
@@ -19,7 +20,22 @@
 /**
  * Provides an interface to a Minecraft server proxy.
  */
-public interface ProxyServer {
+public abstract class ProxyServer {
```

Comment:
Uh, what? You should be able to pass a `ProxyServer` instance wherever you should need it. This doesn't need to be turned into a singleton.

Reply:
> 
> 
> Uh, what? You should be able to pass a `ProxyServer` instance wherever you should need it. This doesn't need to be turned into a singleton.

Gonna fix it, when will deal with code style issues :(